x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/rule_downloader/index.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { exportRules, Rule } from '../../../../../containers/detection_engine/rules';
+import { displayErrorToast, useStateToaster } from '../../../../../components/toasters';
+import * as i18n from './translations';
+
+const InvisibleAnchor = styled.a`
+  display: none;
+`;
+
+export interface RuleDownloaderProps {
+  filename: string;
+  rules?: Rule[];
+  onExportComplete: (exportCount: number) => void;
+}
+
+/**
+ * Component for downloading Rules as an exported .ndjson file. Download will occur on each update to `rules` param
+ *
+ * @param filename of file to be downloaded
+ * @param payload Rule[]
+ *
+ */
+export const RuleDownloaderComponent = ({
+  filename,
+  rules,
+  onExportComplete,
+}: RuleDownloaderProps) => {
+  const anchorRef = useRef<HTMLAnchorElement>(null);
+  const [, dispatchToaster] = useStateToaster();
+
+  useEffect(() => {
+    let isSubscribed = true;
+    const abortCtrl = new AbortController();
+
+    async function exportData() {
+      if (anchorRef && anchorRef.current && rules != null) {
+        try {
+          const exportResponse = await exportRules({
+            ruleIds: rules.map(r => r.rule_id),
+            signal: abortCtrl.signal,
+          });
+
+          if (isSubscribed) {
+            // @ts-ignore function is not always defined -- this is for supporting IE
+            if (window.navigator.msSaveOrOpenBlob) {
```

Comment:
Can we do: `if (isFunction(window.navigator.msSaveOrOpenBlob)) {` and avoid the `@ts-ignore`?

Reply:
Oooh good one -- thanks! 🙂